src/cmd/turnkey/pkg/generate.go

```diff
@@ -0,0 +1,114 @@
+package pkg
+
+import (
+	"encoding/hex"
+
+	"github.com/rotisserie/eris"
+	"github.com/spf13/cobra"
+
+	"github.com/tkhq/go-sdk/pkg/apikey"
+	"github.com/tkhq/go-sdk/pkg/store/local"
+
+	"github.com/tkhq/go-sdk/pkg/enclave_encrypt"
+)
+
+// Filepath to write the target public key generated by `generate encryption-key`.
+var targetPublicKeyPath string
+
+func init() {
+	encryptionKeyCmd.Flags().StringVar(&targetPublicKeyPath, "target-public-key-path", "", "filepath to write the target public key to.")
+
+	generateCmd.AddCommand(apiKeyCmd)
+	generateCmd.AddCommand(encryptionKeyCmd)
+
+	rootCmd.AddCommand(generateCmd)
+}
+
+// generateCmd represents the base command for generating different kinds of keys
+var generateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generate keys",
+}
+
+// Represents the command to generate an API key
+var apiKeyCmd = &cobra.Command{
+	Use:   "api-key",
+	Short: "Generate a Turnkey API key",
+	Long:  `Generate a new API key that can be used for authenticating with the API.`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if Organization == "" {
+			OutputError(eris.New("please supply an organization ID (UUID)"))
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		name, err := cmd.Flags().GetString("key-name")
+		if err != nil {
+			OutputError(eris.Wrap(err, "failed to read API key name"))
+		}
+
+		apiKey, err := apikey.New(Organization)
+		if err != nil {
+			OutputError(eris.Wrap(err, "failed to create API keypair"))
+		}
+
+		if name == "-" {
+			Output(map[string]string{
+				"publicKey":  apiKey.TkPublicKey,
+				"privateKey": apiKey.TkPrivateKey,
+			})
+		}
+
+		apiKey.Metadata.Name = name
+		apiKey.Metadata.PublicKey = apiKey.TkPublicKey
+		apiKey.Metadata.Organizations = []string{Organization}
+
+		if err = keyStore.Store(name, apiKey); err != nil {
```

Comment:
should we store the type (e.g. `AUTHENTICATION` vs `ENCRYPTION`) somewhere so we can make sure we don't mix the use of these keys?

Reply:
ah also, this reminds me i forgot to delete `generate encryption-key` from this PR. i originally added it when i planned on including both import and export in one PR, but then decided to decouple them as export has some additional necessary changes (key store updates, enclave encrypt client updates).

deleting this in 0e6aa6ccf575da4ddb0763066447abc60f488c10